src/sources/datadog/agent/mod.rs

```diff
@@ -209,21 +228,25 @@ impl DatadogAgentSource {
         events: Result<Vec<Event>, ErrorMessage>,
         acknowledgements: bool,
         mut out: SourceSender,
+        output: Option<&str>,
     ) -> Result<Response, Rejection> {
         match events {
             Ok(mut events) => {
                 let receiver = BatchNotifier::maybe_apply_to_events(acknowledgements, &mut events);
 
                 let mut events = futures::stream::iter(events);
-                out.send_all(&mut events)
-                    .map_err(move |error: crate::source_sender::ClosedError| {
-                        // can only fail if receiving end disconnected, so we are shutting down,
-                        // probably not gracefully.
-                        error!(message = "Failed to forward events, downstream is closed.");
-                        error!(message = "Tried to send the following event.", %error);
-                        warp::reject::custom(ApiError::ServerShutdown)
-                    })
-                    .await?;
+                if output.is_none() {
+                    out.send_all(&mut events).await
+                } else {
+                    out.send_all_named(output.unwrap(), &mut events).await
+                }
```

Comment:
This would be a nice place to use an if-let:

```rust
if let Some(name) = output {
    out.send_all_named(name, &mut events).await;
} else {
    out.send_all(&mut events).await
}
```